lib/services/upstart.c

```diff
@@ -61,55 +61,70 @@ upstart_cleanup(void)
     }
 }
 
-static gboolean
-upstart_job_by_name(const gchar * arg_name, gchar ** out_unit, int timeout)
+/*!
+ * \internal
+ * \brief Get the DBus object path corresponding to a job name
+ *
+ * \param[in]  arg_name  Name of job to get path for
+ * \param[out] path      If not NULL, where to store DBus object path
+ * \param[in]  timeout   Give up after this many seconds
+ *
+ * \return true if object path was found, false otherwise
+ * \note The caller is responsible for freeing *path if it is non-NULL.
+ */
+static bool
+object_path_for_job(const gchar *arg_name, char **path, int timeout)
 {
-/*
-  com.ubuntu.Upstart0_6.GetJobByName (in String name, out ObjectPath job)
-*/
+    /*
+        com.ubuntu.Upstart0_6.GetJobByName (in String name, out ObjectPath job)
+    */
     DBusError error;
     DBusMessage *msg;
     DBusMessage *reply = NULL;
     const char *method = "GetJobByName";
+    bool rc = false;
 
-    if(upstart_init() == FALSE) {
-        return FALSE;
+    if (path != NULL) {
+        *path = NULL;
+    }
+
+    if (!upstart_init()) {
+        return false;
     }
     msg = dbus_message_new_method_call(BUS_NAME, // target for the method call
                                        BUS_PATH, // object to call on
                                        UPSTART_06_API, // interface to call on
                                        method); // method name
```

Comment:
It is alway "GetJobByName" - right? So we could hardcode it here to make it clear it is always the same.

Reply:
makes sense